MailCore/Models/Signature.swift

```diff
@@ -60,9 +56,34 @@ public class Signature: Object, Codable, Identifiable {
         // Property 'hash' already exists
         case hashString = "hash"
     }
+
+    override public var hash: Int {
+        id
+    }
+
+    override public func isEqual(_ object: Any?) -> Bool {
```

Comment:
Why do we need custom `hash` and `isEqual` implementation ?

Reply:
The [implementation of the "isEqual" method on the Realm side](https://github.com/realm/realm-swift/blob/354c7ad1827c3f2ec4b51da466c73926984ec245/Realm/RLMObjectBase.mm#LL483C4-L483C4) takes in the realm attached, the detached state and more.

Which is correct, but kind of breaks with our pattern using `.freezeIfNeeded` when I care about something existing or not in a set for a primary key.

Another more verbose way is to filter collections explicitly comparing only on the \id but I'm not sure I like it either.

I guess I can refactor to make sure all the realm objects are both detached in order to _pass_ comparision.

Reply:
Confirmed in DM I'm keeping it this way.